### DISCUSSION TITLE
Connection failures Xdrip app after Samsungs Galaxy A53's latest security patch 01-06-2023

### DISCUSSION THREAD

**Discussion by GDongelmans** [NightscoutFoundation/xDrip#2905] [Q&A]
Hello all first of all im new here on the forum so if i made a mistake i'm sorry. I tried to find the awnser but i couldn't find it. 

The problem we are facing right now is that with the latest android security patch on the Samsung Galaxy A53 5G (with release date 01-06-2023) the connection with xdrip is failing dramaticly. Sometimes it can't connect for up to 30 min straight.  With the closed loop system we are using this is a major problem. It involves the Baseband-version A536BXXU6CWE. The old version A53BXXU5CWD1 worked fine. but a back roll from firmware 6 to 5 is a hard thing to do.

Xdripps doesn't give anny errors. It just loses connection for long intervals. This results in no glucose value.

I attatched a picture of the software data from the phone hope this helps.

![telefoon gegevens](https://github.com/NightscoutFoundation/xDrip/assets/137838035/a14ff0ec-399d-466a-bb4e-f059412a78c8)

**Comment by Navid200** [comment]
After Android update, some Android settings are set back to their default settings.
I have explained this here:
https://navid200.github.io/xDrip/docs/FAQ/AndroidUpdate.html

Please let me know if either of those has been reenabled by the update.

**Reply by GDongelmans** [reply]
Hello first of all thx for the quick responds. One of those settings were reenabled but we set both to unlimited/unrestricted but still the problem stays. At the moment he cant connect at all.

**Reply by Navid200** [reply]
Please post a screenshot of the system status page.

**Reply by GDongelmans** [reply]
Here you go

![Afbeelding van WhatsApp op 2023-06-27 om 14 09 26](https://github.com/NightscoutFoundation/xDrip/assets/137838035/98a8e2a2-a152-4176-89b8-6854e3bc13df)


![Afbeelding van WhatsApp op 2023-06-27 om 14 07 43](https://github.com/NightscoutFoundation/xDrip/assets/137838035/522d7a03-e95e-4f89-bb78-44771e749b00)

**Reply by Navid200** [reply]
Why is everything blank?
Did you uninstall xDrip?  Why?

**Reply by GDongelmans** [reply]
Because it didnt connect for several days. My wife tried resetting it and she tried the pre-release . There is a facebook group where multiple people had same issues so she tried some stuff.  Last days she tried getting the app working on an old device. But for this we deleted it on the other phone. 


Screenshots are from the last "normal" release of xdrip 

She is dependend on the closes loop system for some times now. So if Xdrip can't connect her other systems won't work.

**Reply by Navid200** [reply]
You can disable collection in xDrip.  There is no need to uninstall: https://navid200.github.io/xDrip/docs/Stop-xDrip.html

Now that you have reinstalled, you need to go over the settings and make sure they are all correct.  Also is Bluetooth on on the phone, and is the location service enabled on the phone? 
https://navid200.github.io/xDrip/docs/G6-Recommended-Settings.html

After you correct all the settings, including all the Android settings, wait 5-10 minutes and post an updated screenshot of the G5/G6/  No need for the classic status page.

**Comment by GDongelmans** [comment]
We will go over the settings again and report a new screenshot when the
time has past.

Op di 27 jun 2023 om 14:24 schreef Navid ***@***.***>:

> You can disable collection in xDrip. There is no need to uninstall:
> https://navid200.github.io/xDrip/docs/Stop-xDrip.html
>
> Now that you have reinstalled, you need to go over the settings and make
> sure they are all correct. Also is Bluetooth on on the phone, and is the
> location service enabled on the phone?
> https://navid200.github.io/xDrip/docs/G6-Recommended-Settings.html
>
> After you correct all the settings, including all the Android settings,
> wait 5-10 minutes and post an updated screenshot of the G5/G6/ No need for
> the classic status page.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NightscoutFoundation/xDrip/discussions/2905#discussioncomment-6293280>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BA3T3U2YNUQDDR7PYGR6GG3XNLGHJANCNFSM6AAAAAAZUYVJ7U>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
Met vriendelijke groet,


Gert Dongelmans

**Reply by GDongelmans** [reply]
For now it seems he works. atleast he found a connection. But before we did de whole reinstall thing he had sometimes connection. But at that time we didn't change the battery setting. 

For now im going to set the topic on solved/answerd if things go south again i will reopen/ repost thx for the quick and good support.

![Afbeelding van WhatsApp op 2023-06-27 om 14 39 30](https://github.com/NightscoutFoundation/xDrip/assets/137838035/9f59e755-2c97-47c3-9810-afaf9934207a)

**Reply by Navid200** [reply]
Just to be sure, have you disabled "Avoid scanning"?
Have you enabled "Special pairing workaround"?

**Reply by GDongelmans** [reply]
Yes we did. And we already facing a problem. as you saw on last screenshot we had a connection for some time but it failed quite soon after.

![Afbeelding van WhatsApp op 2023-06-27 om 14 47 05](https://github.com/NightscoutFoundation/xDrip/assets/137838035/4fc70eaa-5735-42ae-a97a-3cefda7aa79f)

**Reply by Navid200** [reply]
This happen if you haven't enabled special pairing workaround.

Please post a screenshot of the following page.

Settings -> G5/G6/Dex1 Debug Settings

**Reply by Navid200** [reply]
Did you get a pair request?  Did you approve it?

**Reply by Navid200** [reply]
What is the status of the old device?
If there is another device that is trying to connect to the same transmitter, it is possible that it could interfere.

**Reply by GDongelmans** [reply]
We got a pair request and approved it. On the old device we deleted the app( only reason we used it was that it was constant there). We disbabled and reabled(and rebooted)  the work arround for samsung so we are waiting now if he keeps his connections

**Comment by GDongelmans** [comment]
The connection stays up for some time but there is no glucose value 


![Afbeelding van WhatsApp op 2023-06-27 om 15 32 03](https://github.com/NightscoutFoundation/xDrip/assets/137838035/c3a104fa-e6be-44ad-aa04-18bf44a2cd22)
![Afbeelding van WhatsApp op 2023-06-27 om 15 35 15](https://github.com/NightscoutFoundation/xDrip/assets/137838035/f6433ddd-d45e-4da1-b347-ada8bbbe6383)

**Reply by Navid200** [reply]
Last connected shows 11 minutes, which is greater than 5 minutes.  So, there is a connectivity problem.

Have you disabled trust auto-connect on the Bluetooth page?

**Reply by GDongelmans** [reply]
![Afbeelding van WhatsApp op 2023-06-27 om 15 40 59](https://github.com/NightscoutFoundation/xDrip/assets/137838035/3daa6747-0ca4-49bf-92d5-e8ff7f5fc6ce)


This one is off

**Reply by Navid200** [reply]
Is last connected still greater than 5 minutes?

**Reply by GDongelmans** [reply]
Sadly it is

![Afbeelding van WhatsApp op 2023-06-27 om 15 54 45](https://github.com/NightscoutFoundation/xDrip/assets/137838035/b6fdf71b-4dd9-475c-87e4-bc6cb8702d72)

**Reply by Navid200** [reply]
Is the transmitter paired?
If you go to Android settings and look for paired devices or previously paired devices?
Do you see DexcomCE on the list?

**Reply by GDongelmans** [reply]
Yes it is paired


![afbeelding_2023-06-27_160853865](https://github.com/NightscoutFoundation/xDrip/assets/137838035/52843e0e-a829-42c3-9a49-a4e6b0c55100)

**Reply by Navid200** [reply]
Please restart the phone, and wait 5 minutes.  Does last connected update to a value less than 5 minutes?

**Reply by GDongelmans** [reply]
![Afbeelding van WhatsApp op 2023-06-27 om 16 26 15](https://github.com/NightscoutFoundation/xDrip/assets/137838035/18869c1f-c8de-4149-9425-6c0d8e89af11)


no luck

**Reply by Navid200** [reply]
What do the logs show?

**Comment by GDongelmans** [comment]
![Afbeelding van WhatsApp op 2023-06-27 om 16 35 58](https://github.com/NightscoutFoundation/xDrip/assets/137838035/cd259685-af9c-49a5-a80a-d6a80956e250)

**Comment by Navid200** [comment]
@jamorham

**Reply by Navid200** [reply]
While we wait for JamOrHam to respond, do you happen to have a Dexcom receiver?  If yes, you can use it with no interference with xDrip.

**Comment by GDongelmans** [comment]
Sadly no but we can use an old device to make The connection for the time
being. This time we will keep it installed And stop The transfer on the
normal Phone.

Age did try to connect with dexcom app but it would't connect either.
Although it wouldnt help because we need androidapps 2but still maby its
usefull info for you.

Thx for the help and guidance so far.

Op di 27 jun. 2023 17:16 schreef Navid ***@***.***>:

**Comment by jamorham** [comment]
Steps to try to resolve the issue:

1) Update to xDrip latest nightly version.
2) Enable OB1 debugging: https://github.com/NightscoutFoundation/xDrip/wiki/Enabling-Debugging-for-OB1-collector
3) To check if it is an issue with phone doing power-saving, stay on the G6 status page - The screen will stay on during this test - Does it still have a problem here or only when the screen is off? If the problem is when the screen is off then it points to power saving issues, otherwise we have to look elsewhere. - If so please use the log upload feature so I can take a look.

**Reply by GDongelmans** [reply]
We did everything you said And made an upload of The log sience The update
there is no connection checked battery premisions And work arround settings

**Comment by Navid200** [comment]
@GDongelmans 
Please follow any advice coming from JamOrHam.  I am going to be quiet now.

But, would you please not respond to posts using your email.  Please instead log into GitHub and respond directly.  
Otherwise, the thread becomes hard to follow for anyone including JamOrHam, whose time is limited compared to mine.

**Reply by GDongelmans** [reply]
I will in the future my bad. Thanks for the help.

**Comment by jamorham** [comment]
The logs need more time to load data so if it has been running please upload again. Thanks

**Reply by GDongelmans** [reply]
I did a New upload there wereld sometimes hé could connect do it should be more usefull data

**Comment by jamorham** [comment]
I am not sure from what you say whether it is working good? You can also try turning off the "special pairing workaround" feature and see if that changes it for the better or not?

**Reply by GDongelmans** [reply]
Sometimes we get some data but normaly hé generates once every 5 min. Now it sometimes takes up to 45 min.  So it is very unstable And unreliable. But we will try turning off The setting

![IMG-20230627-WA0023.jpg](https://github.com/NightscoutFoundation/xDrip/assets/137838035/3ea29d31-637f-4de7-9629-52e34410dd88)

**Reply by GDongelmans** [reply]
We disabeld the feature of the workaround but it doenst help annything

**Comment by bravobip1** [comment]
Hello, many of us we have the same problem with the latest android security patch of samsung. Something is blocking the bluetooth and it can't ''see'' the sensor.

**Reply by GDongelmans** [reply]
Yeah my wife(the user) is part of a community on facebook who shares similair problems since the june patch.

**Comment by hanbozdugan** [comment]
I'm quite a tech literate person and in my opinion the problem is not in the xDrip+ settings but in the latest system update of Samsung phones. In my case A52 and A53. Yesterday we launched the official Samsung A53 app and it first said to wait 40 minutes for it to connect. It connected but after a while the connection was lost. I assume that Samsung put some kind of protection on the bluetooth or changed something on the drivers of the bluetooth itself. Now I seem to have opened a job for you to hack and that. I guess Dexcom should also work on this problem now

**Reply by Navid200** [reply]
This is not new.  Samsung Bluetooth stack has been different.
Our lead developer has had to spend a lot of time tweaking xDrip so that people can use Samsung Considering it is a popular phone.

He has a Samsung and will test this.
If you want to help, please see and follow this:
https://github.com/NightscoutFoundation/xDrip/discussions/2915

**Reply by erikaasen** [reply]
I can confirm this is most likely not a xDrip problem. I have waited, but did the update on my sons Samsung A33 last night before we go on one week vacation from tomorrow. xDrip and AndroidAPS instantly stopped working.

Just get a few bs updates now and then. So I need a fix now ASAP today. I guess I just need to buy a new phone? Had issues last summer with OnePlus Nord after an update (Stopped working after 40min and at night because of their agressive battery saving features I believe). Changed to the A33, worked perfectly, but now this.

So I am considering buying the Pixel 7a right now today. Anyone have any experience with this phone? Any other solutions than replacing the Samsung?? Thanks..

**Reply by Navid200** [reply]
@erikaasen I have had my very first experience with a Pixel phone over the past few months, Pixel6a.
It is unbelievably solid with respect to Bluetooth connectivity with a G6.

The only unexpected experience I have had is that there is a Google bar at the bottom of the main screen that cannot be removed unless you install an unofficial launcher.   I've got used to it by now.

**Reply by erikaasen** [reply]
@Navid200 thanks for your reply. I ended up bying a Motorola G73 for half the price of the Pixel 7a. So far it is performing very stable both with xDrip running a Dexcom G6 and AndroidAPS with a Omnipod Dash.

**Comment by PhilTU** [comment]
> I'm quite a tech literate person and in my opinion the problem is not in the xDrip+ settings but in the latest system update of Samsung phones. In my case A52 and A53. 

Users of other Libre apps including diabox and Abbott's Librelink are also having problems. One guy switched to his old Samsung 21 and it worked instantly.

**Comment by xamrex** [comment]
I m using freestyle Libre 2, and also after samusng system update (A536BXXU6CWE9) I got no data from bluetooth ;-(

**Comment by PhilTU** [comment]
![image](https://github.com/NightscoutFoundation/xDrip/assets/76240903/e9c49f59-45a8-4452-abdb-32e1e4c9bcae)

possible solution on the horizon ?

**Comment by caocosta** [comment]
The problem for the Samsung not connecting to the bluetooth of the libre 2 seems to not be the phone it self but the app xDrip.
Have you tried to connect to libre 2 using the app Juggluco?
Well, I tried, and it worked.
So the problem is not my Samsung A52s bluetooth because it works fine with the app Juggluco, but it refuses to connect with the app xDrip.
I tried a million times with all possible xDrip configuration combinations and waited millions of seconds for a successful connection, and nothing!!!
But Juggluco connects to libre 2 at first attempt with no effort at all. With same phone.
Only it is a worse app with much less options.... 
What do you say?
I'm very disapointed with xDrip cause it always worked well with libre1 and miao miao or blucon, but now with libre2 I never got it to work, neither with blucon (it connects but gives wrong values) nor directly with libre 2 (it won't connect to bluetooth). I had to fall back to use the much worse official  blucon apps, Linkblucon and Followblucon. Much much worse than xDrip :( , but they work.
Waiting for an xDrip update that solves this... 
Greetings...

**Reply by PhilTU** [reply]
Are you in the US? US Libre 2 sensor is not supported in xDrip+

With OOP2 and compatible sensors xDrip works well with Libre 2.

**Comment by xamrex** [comment]
its the samsung.
Few days ago they released update for my A53, and now its working.

**Reply by GDongelmans** [reply]
It does. as stated before the  security patch from samsung in juni messed up the bluetooth connection. With the latest patch from july it is fixed. but it depends when you get the patch. Libre had the same problem and they sat down with samsung to fix it. So thank you guys from Xdrip for helping ppl get a loop system without the much needed finances .
